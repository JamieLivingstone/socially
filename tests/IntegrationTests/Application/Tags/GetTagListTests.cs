using System.Linq;
using System.Threading.Tasks;
using Application.Common.Exceptions;
using Application.Tags.Queries.GetTagList;
using FluentAssertions;
using IntegrationTests.Application.TestUtils;
using NUnit.Framework;
using Snapshooter.NUnit;

namespace IntegrationTests.Application.Tags;

[TestFixture]
public class GetTagListTests : TestBase
{
  [Test]
  public void GivenAnInvalidRequest_ThrowsValidationException()
  {
    var query = new GetTagListQuery
    {
      PageSize = int.MaxValue // Exceeds page size constraint
    };

    FluentActions.Invoking(() => SendAsync(query)).Should().ThrowAsync<ValidationException>();
  }

  [Test]
  public async Task GivenAPaginatedRequest_ReturnsTagList()
  {
    var query = new GetTagListQuery
    {
      PageNumber = 1,
      PageSize = 2,
    };

    var result = await SendAsync(query);

    Snapshot.Match(result);
  }

  [Test]
  public async Task GivenAPaginatedRequestWithTerm_ReturnsTagList()
  {
    var query = new GetTagListQuery
    {
      PageNumber = 1,
      PageSize = 10,
      Term = Seed.Tags().First().TagId
    };

    var result = await SendAsync(query);

    Snapshot.Match(result);
  }
}
