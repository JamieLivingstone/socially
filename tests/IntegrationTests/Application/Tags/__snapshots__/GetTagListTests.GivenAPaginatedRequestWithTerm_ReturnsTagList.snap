{
  "Items": [
    {
      "Name": "tag-one"
    }
  ],
  "PageNumber": 1,
  "TotalPages": 1,
  "TotalCount": 1,
  "HasPreviousPage": false,
  "HasNextPage": false
}
